docs/src/examples/sensitivity-analysis-ridge.jl

```diff
@@ -64,76 +64,104 @@ function fit_ridge(X, Y, alpha = 0.1)
     set_silent(model)
     @variable(model, w) # angular coefficient
     @variable(model, b) # linear coefficient
-    @variable(model, e[1:N]) # approximation error
-    ## constraint defining approximation error
-    @constraint(model, cons[i=1:N], e[i] == Y[i] - w * X[i] - b)
+    ## expression defining approximation error
+    @expression(model, e[i=1:N], Y[i] - w * X[i] - b)
     ## objective minimizing squared error and ridge penalty
     @objective(
         model,
         Min,
-        dot(e, e) + alpha * (sum(w * w) + sum(b * b)),
+        1 / N * dot(e, e) + alpha * (w^2 + b^2),
     )
     optimize!(model)
-    return model, w, b, cons # return model, variables and constraints references
+    return model, w, b # return model & variables
 end
 
 
-# Train on the data generated.
+# Plot the data points and the fitted line for different alpha values
 
-model, w, b, cons = fit_ridge(X, Y)
-ŵ, b̂ = value(w), value(b)
-
-# We can visualize the approximating line.
-
-p = Plots.scatter(X, Y, label="")
+p = Plots.scatter(X, Y, label=nothing, legend=:topleft)
 mi, ma = minimum(X), maximum(X)
-Plots.plot!(p, [mi, ma], [mi * ŵ + b̂, ma * ŵ + b̂], color=:red, label="")
+Plots.title!("Fitted lines and points")
 
+for alpha in 0.5:0.5:1.5
+    local model, w, b = fit_ridge(X, Y, alpha)
+    ŵ = value(w)
+    b̂ = value(b)
+    Plots.plot!(p, [mi, ma], [mi * ŵ + b̂, ma * ŵ + b̂], label="alpha=$alpha", width=2)
+end
+p
 
 # ## Differentiate
 
 # Now that we've solved the problem, we can compute the sensitivity of optimal
-# values of the angular coefficient `w` with
+# values of the slope `w` with
 # respect to perturbations in the data points (`x`,`y`).
 
-# Begin differentiating the model.
-# analogous to varying θ in the expression:
-# ```math
-# e_i = (y_{i} + \theta_{y_i}) - w (x_{i} + \theta_{x_{i}}) - b
-# ```
+alpha = 0.4
+model, w, b = fit_ridge(X, Y, alpha)
+ŵ = value(w)
+b̂ = value(b)
+
+# We first compute sensitivity of the slope with respect to a perturbation of the independent
+# variable `x`.
+
+# Recalling that the points $(x_i, y_i)$ appear in the objective function as:
+# `(yi - b - w*xi)^2`, the `DiffOpt.ForwardInObjective` attribute must be set accordingly,
+# with the terms multiplying the parameter in the objective.
+
+∇x = zero(X)
+∇y = zero(X)
 
-∇ = zero(X)
 for i in 1:N
-    for j in 1:N
-        MOI.set(
-            model,
-            DiffOpt.ForwardInConstraint(),
-            cons[j],
-            i == j ? index(w) + 1.0 : 0.0 * index(w)
-        )
-    end
+    # x sensitivity
+    MOI.set(
+        model,
+        DiffOpt.ForwardInObjective(),
+        -(w^2 * X[i] + 2b * w - 2 * w * Y[i])
```

Comment:
also I am not 100% with those forward input sensitivities
because there is a squared input parameter (`X[I]`)

Reply:
sensitivities are exaclty the same as in the previous formulation?
It is not clear to me that the implemented method accepts squared parameters, and if so, what changes is not clear to me at least

Reply:
I started drafting this:
```
min_x    a^2 * x^2 + x

where a is parameter and x is a variable

sol is:
2xa^2 + 1 = 0
=> x = 1/(2a^2)

so dx/da = -1 / (a^3)

while

min_x    a * x^2 + x

where a is parameter and x is a variable

sol is:
2xa + 1 = 0
=> x = 1/(2a)

so dx/da = -1 / (2a^2)
```

We can write a very simple tutorial with this example and compare results.
I think we might not be able to obtain the first from DiffOpt, since the paper assume that coefficient depend linearly on parameters.

Reply:
For a simplified problem (alpha = 0 and b = 0) we can compare with the analytical derivative.
```
f(w) = \sum_i(y_i-w*x_i)^2 = \sum_i(y_i^2 - 2*w*x_i*y_i + w^2*x_i^2)

min_w f(w)

min_w \sum_i(y_i^2 - 2*w*x_i*y_i + w^2*x_i^2)

df/dw = 0 

=>

0 = \sum_i(-2*x_i*y_i + 2w*x_i^2)

=> 

0 = -2*\sum_i(x_i*y_i) + 2w*\sum_i(x_i^2)

=>

w = \sum_i(x_i*y_i) / \sum_i(x_i^2)

w(x_i) = (x_i*y_i + B) / (x_i^2 + A)

where

A = \sum_{j st j != i}(x_j*y_j)
B = \sum_{j st j != i}(x_j*x_j)

dw/dx_i = (y_i*(A - x_i^2) - 2*B*x_i) / (x_i^2 + A)^2

```